tools/graphiteindex/src/main/java/com/criteo/biggraphite/graphiteindex/LuceneIndexSearcher.java

```diff
@@ -0,0 +1,242 @@
+                                     BooleanQuery luceneQuery,
+                                     ColumnDefinition column) {
+            this.ssTables = ssTables.iterator();
+            this.readCommand = readCommand;
+            this.query = luceneQuery;
+            this.column = column;
+        }
+
+        public boolean hasNext() {
+            if(currentResults != null && currentResults.hasNext()) {
+                return true;
+            } else {
+                // Initialize new search results
+                while(ssTables.hasNext()) {
+                    currentSSTable = ssTables.next();
+                    String indexName = GraphiteSASI.makeIndexName(column, currentSSTable.descriptor.generation);
+                    Path indexPath = new File(currentSSTable.descriptor.directory, indexName).toPath();
+                    logger.debug("Searching ssTable:{} with indexPath:{}", currentSSTable.getFilename(), indexPath);
+                    List<Long> offsets = searchOffsets(query, indexPath);
+                    currentResults = offsets.iterator();
+                    if(currentResults.hasNext()) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        public UnfilteredRowIterator next() {
+            try {
+                long indexPosition = currentResults.next();
+                DecoratedKey decoratedKey = currentSSTable.keyAt(indexPosition);
+                logger.debug("Fetching partition at indexPosition:{} from SSTable:{}",
+                        indexPosition, currentSSTable.getFilename());
+
+                UnfilteredRowIterator uri = new UnfilteredRowIteratorWithLowerBound(decoratedKey,
+                        currentSSTable,
+                        readCommand.clusteringIndexFilter(decoratedKey),
+                        readCommand.columnFilter(),
+                        readCommand.isForThrift(),
+                        readCommand.nowInSec(),
+                        false,
+                        SSTableReadsListener.NOOP_LISTENER);
+                currentSSTable.incrementReadCount();
+                return uri;
+            } catch(IOException e) {
+                logger.error("Exception while fetching partition", e);
+            }
+            return null;
+        }
+
+        public List<Long> searchOffsets(BooleanQuery query, Path indexPath)
+        {
+            return search(query, indexPath, LuceneUtils::getOffsetFromDocument);
+        }
+
+        /**
+         * TODO(p.boddu): Lot of heavy lifting here for each query. Optimize!
+         */
+        private <T> List<T> search(BooleanQuery query, Path indexPath, Function<Document, T> handler)
+        {
+            logger.info("{} - Searching for generated query: {}", indexPath, query);
+
+            ArrayList<T> results = new ArrayList<>();
+            Collector collector = new MetricsIndexCollector(
```

Comment:
If you want to rename every other MetricsIndex to LuceneIndex then please rename MetricsIndexCollector as well for consistency.

Reply:
You got it.